mlir/lib/Dialect/Linalg/TransformOps/LinalgTransformOps.cpp

```diff
@@ -98,12 +98,34 @@ static DiagnosedSilenceableFailure unpackSingleIndexResultPayloadOperations(
       result.push_back(ofr);
       continue;
     }
-    auto payloadOps = state.getPayloadOps(ofr.get<Value>());
+
+    Value transformValue = ofr.get<Value>();
+    if (isa<ParamType>(transformValue.getType())) {
+      ArrayRef<Attribute> params = state.getParams(transformValue);
+      if (!isa<IntegerAttr>(params[0]))
+        return transformOp.emitDefiniteFailure() << "expected IntegerAttr";
+      result.push_back(params[0]);
+      continue;
+    }
+    if (isa<AnyParamType>(transformValue.getType())) {
+      ArrayRef<Attribute> params = state.getParams(transformValue);
+      if (!isa<ArrayAttr>(params[0]))
+        return transformOp.emitDefiniteFailure() << "expected ArrayAttr";
+      ArrayAttr paramsArray = cast<ArrayAttr>(params[0]);
+      for (Attribute param : paramsArray.getValue()) {
+        if (!isa<IntegerAttr>(param))
+          return transformOp.emitDefiniteFailure() << "expected IntegerAttr";
+        result.push_back(param);
```

Comment:
It's unclear to me what is the intended model here. Looking at the code, if the type is `any_param`, it will accept array-of-integer parameters and expand its context as tile sizes. But it will keep doing so for other parameters, so we may end up having more sizes than loops, and a surprising mismatch between parameter positions and loop depth. We could accept *one* parameter, and unpack it, but I don't think we want to accept multiple here. Also, there's no need to rely on array attributes as a parameter can be associated with a list of attributes (unless we want to support some multi-level scheme where the parameter is a list-of-lists of tile sizes to use for each of the loops associated with the input).

Reply:
I don't believe this was addressed.

Reply:
I thought moving the support of any_param to the second `unpackSingleIndexResultPayloadOperations` that takes a `Value` would address it since there won't be lists inside lists. Could you give me more guidence on this?

Reply:
As far as I recall, there are currently two modes of functioning here:
- there are N unpacked handles, where each handle corresponds to one dimension; each handle points to the same number of operations as the "target" handle;
- there is 1 packed handle, which points to as many operations as dimensions; same dimensions are used for tiling all operations associated with the "target" handle.

Parameters should behave similarly. Since parameters are associated with a _list_ of attributes, we would treat that list identically to how we treat the list of operations for operation handle. Consequently, we should never expect an array attribute in parameters. There is already the implicit list and we must use that instead.

Reply:
Please review, thanks!